### PR TITLE
Replace local tunnel with automated google function deployment (only fo CI)

### PR DESCRIPTION
**Pull request by ahmetoz**
**Situation**
Local tunnel is used to exposes the localhost to internet for testing purposes as API extension trigger url expects a valid URL to trigger.
Currently it's used on the integration and e2e tests, and creating a tunnel before the tests started. See: https://github.com/commercetools/commercetools-adyen-integration/blob/master/extension/test/integration/integration-test-set-up.js#L65-L80

**Complication**
Local tunnel is working well for development purposes (e.g debugging) but does not simulate the reality of the integration. As extension is running on a local node server, but it should be running either in a container or in a serverless function such as google functions. Also we faced some issues with localtunnel on tests that is being flaky (resolved but might cause another issue on the future)

**Resolution**
Replace local tunnel such as using google function function URL as an API extension trigger url on test fixtures.

------

# Review notes: 
- First challange was doing automated deployment to GCP, it's resolved with gcloud sdk (see: ci.yaml changes) 
- Second challange was not removing resources before and after the tests to make the test scenarios more realistic.
- Local tunnel is useful for development (debugging purpose) so it's the default hook if the CI env variable is not set ( github actions sets this variable to true so CI flow uses GCP functions on test fixtures)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -22,9 +24,33 @@ jobs:
...
-      - run: npm run test
-        env:
-          ADYEN_INTEGRATION_CONFIG: ${{ secrets.ADYEN_INTEGRATION_CONFIG }}
+      - run: npm run check
+      - run: npm run cover:unit
+      - name: prepare extensions folder
+        run: |
+          echo $ADYEN_INTEGRATION_CONFIG >> .extensionrc
+          mkdir -p sources && cp index.googleFunction.js index.js
+          cp -R {index.js,.extensionrc,src,resources,package.json} sources
```
Comment (lojzatran) [inline .github/workflows/ci.yml:33]: I checked README in deployment examples for google functions and `extensionrc` is not mentioned there, though it is used in `zip-google-function` script. Can you update the README?

**Review by lojzatran — CHANGES_REQUESTED** [review]
How long does it take now to run the build?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ahmetoz) [reply .github/workflows/ci.yml:33]: https://github.com/commercetools/commercetools-adyen-integration/pull/890/commits/6fec71b11d35148d60adf4b957b087242c342c8a

**Comment by ahmetoz** [comment]
> How long does it take now to run the build?

6-7 min, a comment from the other PR description, which contains both changes (extension + notification):
> Overall CI build time is increased around 4 minutes because of GCP function deployment, now we have around 6-7 minutes on build.

File: deployment-examples/aws/serverless/extension/README.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ## Deploy as AWS Lambda
 
 1. In order to setup the extension module config, follow our [how to run guide](../../../../extension/docs/HowToRun.md).
-2. Extension module requires `ADYEN_INTEGRATION_CONFIG` as `environment variable`.
+2. Extension module requires `ADYEN_INTEGRATION_CONFIG` as `environment variable` `.extensionrc` file to configure with [external file configuration](../../../../extension/docs/HowToRun.md#external-file-configuration).
```
Comment (lojzatran) [inline deployment-examples/aws/serverless/extension/README.md:4]:
```suggestion
2. Extension module requires `ADYEN_INTEGRATION_CONFIG` as `environment variable` or `.extensionrc` file to configure with [external file configuration](../../../../extension/docs/HowToRun.md#external-file-configuration).
```

File: deployment-examples/gcp/serverless/notification/README.md [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ## Deploy as GCP Cloud Function
 
 1. In order to setup the notification module config, follow our [how to run guide](../../../../notification/docs/HowToRun.md).
-2. Notification module requires `ADYEN_INTEGRATION_CONFIG` as `Runtime environment variable`.
+2. Notification module requires `ADYEN_INTEGRATION_CONFIG` as `Runtime environment variable` `.notificationrc` file to configure with [external file configuration](../../../../notification/docs/HowToRun.md#external-file-configuration).
```
Comment (lojzatran) [inline deployment-examples/gcp/serverless/notification/README.md:4]:
```suggestion
2. Notification module requires `ADYEN_INTEGRATION_CONFIG` as `Runtime environment variable` or `.notificationrc` file to configure with [external file configuration](../../../../notification/docs/HowToRun.md#external-file-configuration).
```

**Review by lojzatran — APPROVED** [review]

**Pull request merged**